package/network/services/lldpd/files/lldpd.init

```diff
@@ -19,17 +26,85 @@ find_release_info()
 	echo "${PRETTY_NAME:-Unknown OpenWrt release} @ $(cat /proc/sys/kernel/hostname)"
 }
 
+get_config_restart_hash() {
+	local var="$1"
+	local _string _hash v
+
+	config_load 'lldpd'
+
+	config_get      v 'config' 'lldp_class'; append _string "$v" ","
+	config_get      v 'config' 'agentxsocket'; append _string "$v" ","
+	config_get      v 'config' 'cid_interface'; append _string "$v" ","
+	config_get      v 'config' 'filter'; append _string "$v" ","
+	config_get_bool v 'config' 'readonly_mode'; append _string "$v" ","
+	config_get_bool v 'config' 'lldpmed_no_inventory'; append _string "$v" ","
+	config_get_bool v 'config' 'lldp_no_version'; append _string "$v" ","
+
+	config_get_bool v 'config' 'enable_lldp' 1; append _string "$v" ","
+	config_get_bool v 'config' 'force_lldp'; append _string "$v" ","
+
+	config_get_bool v 'config' 'enable_cdp'; append _string "$v" ","
+	config_get      v 'config' 'cdp_version'; append _string "$v" ","
+	config_get_bool v 'config' 'force_cdp'; append _string "$v" ","
+	config_get_bool v 'config' 'force_cdpv2'; append _string "$v" ","
+
+	config_get_bool v 'config' 'enable_edp'; append _string "$v" ","
+	config_get_bool v 'config' 'force_edp'; append _string "$v" ","
+
+	config_get_bool v 'config' 'enable_fdp'; append _string "$v" ","
+	config_get_bool v 'config' 'force_fdp'; append _string "$v" ","
+
+	config_get_bool v 'config' 'enable_sonmp'; append _string "$v" ","
+	config_get_bool v 'config' 'force_sonmp'; append _string "$v" ","
+
+	_hash=`echo -n "${_string}" | md5sum | awk '{ print \$1 }'`
+	export -n "$var=$_hash"
+}
+
+get_config_cid_ifaces() {
+	local _ifaces
+	config_get _ifaces 'config' 'cid_interface'
+
+	local _iface _ifnames=""
+	for _iface in $_ifaces; do
+		local _ifname=""
+		if network_get_device _ifname "$_iface" || [ -e "/sys/class/net/$_iface" ]; then
+			append _ifnames "${_ifname:-$_iface}" ","
+		fi
+	done
+
+	export -n "${1}=$_ifnames"
+}
+
 write_lldpd_conf()
 {
-	. /lib/functions/network.sh
-
 	local lldp_description
+	local lldp_hostname
+	local lldp_platform
+	local lldp_tx_interval
+	local lldp_tx_hold
+	local lldp_mgmt_ip
+	local lldp_agetntype
```

Comment:
This PR carried over the same mistakes as @marek22k - see my branch for the relevant corrections:

https://github.com/openwrt/openwrt/compare/main...systemcrash:openwrt:lldp_fixes3

i.e. change to `lldap_agenttype`